jetty-client/src/main/java/org/eclipse/jetty/client/http/HttpSenderOverHTTP.java

```diff
@@ -225,6 +228,10 @@ protected Action process() throws Exception
                         headerBuffer = httpClient.getByteBufferPool().acquire(httpClient.getRequestBufferSize(), false);
                         break;
                     }
+                    case HEADER_OVERFLOW:
+                    {
+                        throw new BadMessageException(INTERNAL_SERVER_ERROR_500, "Request header too large");
```

Comment:
On the client `INTERNAL_SERVER_ERROR_500` won't make much sense, so perhaps replace with `IllegalArgumentException`?
Also, `headerBuffer` should be released to the pool.

Reply:
The BadMessageException was thrown previously, even fit generating a request. Do we really want to change this in 9?

Reply:
@gregw the `header` buffer should be released to the pool.
I'm ok to not change this in 9, but remember to make the change when merging to 10.

Reply:
Ooops missed this one. Fixed now.